### ISSUE TITLE
[Bug] Deepseek V3 on GB200 loss nan

### ISSUE THREAD

**Issue by sophiayyya** [NVIDIA/Megatron-LM#5782]
> **Describe the bug**
> 
> When I refer to this [script](https://github.com/NVIDIA/Megatron-LM/blob/dev/examples/moe_recipes/deepseek_v3/gb200/mxfp8_256GPU_TP1PP4EP64.yaml) to pretrain DeepSeek-V3, I found that the gradient norm becomes NaN with commit [d64ba4c](https://github.com/NVIDIA/Megatron-LM/commit/d64ba4ccb1e3e878c15171c9cc58d5d3b46bf4d5).
> 
> I also tested a reduced-layer version of DeepSeek-V3 with commit [d64ba4c](https://github.com/NVIDIA/Megatron-LM/commit/d64ba4ccb1e3e878c15171c9cc58d5d3b46bf4d5). In that case, the gradient norm is unusually high, and the training loss is not NaN but is significantly higher than expected. Brown line is runned with commit d64ba4ccb1e3e878c15171c9cc58d5d3b46bf4d5. Red line is normal and runned with commit 1b47bc02f7d364fb123535f84ad89c9e110ef8bc.
> 
> <img alt="Image" width="501" height="312" src="https://private-user-images.githubusercontent.com/14706592/620884797-96953d4f-657e-486a-a774-cb6115350ef7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODM5NTE2MTUsIm5iZiI6MTc4Mzk1MTMxNSwicGF0aCI6Ii8xNDcwNjU5Mi82MjA4ODQ3OTctOTY5NTNkNGYtNjU3ZS00ODZhLWE3NzQtY2I2MTE1MzUwZWY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA3MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNzEzVDE0MDE1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE3YTliYzM3ZGI1ZjkxYjM2MWY2OTk3MGEzN2Y0Zjg0NTY2NDA5ZDM0YjY4MDljMTEzYzVhYTJkOGVmZDY2ODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.RXoUy62Zu51PaayM1KjSx_PDmJ61bx4a9uZkWmUuWL0">

> **Steps/Code to reproduce bug**
> 
> Please use this script to reproduce.
> 
> [gb200_2node.sh](https://github.com/user-attachments/files/29969514/gb200_2node.sh) Reproduce the bug with commit [d64ba4c](https://github.com/NVIDIA/Megatron-LM/commit/d64ba4ccb1e3e878c15171c9cc58d5d3b46bf4d5) in dev branch.
> 
> **Expected behavior**
> 
> Loss decreases normally.

**Comment by cspades** [comment]
@YangFei1990 can you take a look at this? It's M-Lite but hopefully not too difficult to debug.

**Comment by Phlip79** [comment]
Does this bug still need investigation @sophiayyya?

**Comment by sophiayyya** [comment]
@Phlip79 I didn't test with the newer code.
